packages/lesswrong/server/callbacks.ts

```diff
@@ -178,6 +181,45 @@ export async function userDeleteContent(user: DbUser, deletingUser: DbUser) {
 
     await commentReportPurgeAsSpam(comment);
   }
+
+  const tags = await Tags.find({userId: user._id}).fetch()
+  // eslint-disable-next-line no-console
+  console.info("Deleting tags: ", tags)
+  for (let tag of tags) {
+    if (!tag.deleted) {
+      try {
+        await updateMutator({
+          collection: Tags,
+          documentId: tag._id,
+          set: {deleted: true},
+          currentUser: deletingUser,
+          validate: false
+        })
+      } catch(err) {
+        // eslint-disable-next-line no-console
+        console.error("Failed to delete tag")
+        // eslint-disable-next-line no-console
+        console.error(err)
+      }
+    }
+  }
+  
+  const revisions = await Revisions.find({userId: user._id}).fetch()
+  // eslint-disable-next-line no-console
+  console.info("Deleting revisions: ", revisions)
+  await Revisions.remove({userId: user._id})
```

Comment:
Minor – should this go after the following loop? It seems like if the operation fails halfway through, then you'll have deleted all the revisions pertaining to this user, and so won't be able to easily run `syncDocumentWithLatestRevision` on subsequent attempts.

Reply:
Oh wait, maybe not, because `syncDocumentWithLatestRevision` assumes that these revisions are already deleted in order to do its thing...